Azure-Samples/Cognitive-Services-Voice-Assistant
# Add better latency management and RealTimeAudio for VA Test Tool

## Purpose
* Add a user perceived latency value.
   * This value is the amount of time since the user stopped speaking until an activity is received.
   * In order to get an accurate value the RealTimeAudio bool must be set to true and Timeout must be set to the duration of the longest wav file to prevent the test tool from timing out. (In the AppConfig)
* Add a flag and properties to simulate x1 speech streaming to Speech SDK
* Update Readme with references to RealTimeAudio and optional WavFile string input
* Turn WavFile contains an optional format where user can define the expected length of speech, this value is updated in the output file with the actual duration of speech received from Speech SDK

## Does this introduce a breaking change?
[ ] Yes
[x] No

## Pull Request Type
What kind of change does this Pull Request introduce?
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:

## How to Test
* In the Appconfig.json add your speech key, region, ```RealTimeAudio: true``` and ```Timeout: 8000```
   * Above Timeout value should be greater than the duration of your longest wav file to prevent the tool from timing out.
* For testing I used a core bot, multi turn [sample files](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/tree/master/clients/csharp-dotnet-core/voice-assistant-test/docs/core-bot-examples/multi-turn)
